proxy/src/registry/transaction.rs

```diff
@@ -293,7 +293,9 @@ where
                 State::Confirmed {
                     block, timestamp, ..
                 } => {
-                    let target = block.checked_add(BLOCK_BOUND).unwrap_or(MIN_CONFIRMATIONS);
+                    let target = block.map_or(0, |b| {
```

Comment:
Can I ask why `0` is safe here?

Reply:
I am not certain it is, thus my comment on the description. My idea is that the block height of a tx starts in 0, so if there is no block for a given tx, treating its height as `0` seems appropriate. @CodeSandwich could you have a look at this as well?

Reply:
This will result in txs always being settled if there is no block number provided..

Reply:
I think that finding `None` in this context is a nasty error and should be treated as such. If I undrestand correctly, this comes from change in the `client::ClientT` API, where `block_header` changed its return type from `Result<BlockHeader>` to `Result<Option<BlockHeader>>`. `None` is returned when the given block does not exist. It was lifted from a variant of the `Error` to a separate layer, because not in all contexts it's an error. In this one I think it is, we're 100% sure that this block should exist.

Reply:
> this comes from change in the client::ClientT API, where block_header changed its return type from Result<BlockHeader> to Result<Option<BlockHeader>>

That is correct, @CodeSandwich. I will add an error layer here then and revert the change to using Options.

Reply:
Done, thanks for the feedback!